azure-cli/sql/deploy.ps1

```diff
@@ -62,7 +62,7 @@ $sqlServerPassword = az keyvault secret show `
 
 if (!$?) {
   Write-Host "  Creating SqlServerPassword secret" -ForegroundColor DarkYellow
-  $sqlServerPassword = Get-NewPassword
+  $sqlServerPassword = New-Password -Length 20
```

Comment:
Length of old function was 16, suggest to keep that

Reply:
https://learn.microsoft.com/en-us/sql/relational-databases/security/strong-passwords?view=sql-server-ver16
>Microsoft SQL Server passwords can contain up to 128 characters, including letters, symbols, and digits

What is a reason good reason to keep new passwords at 16 in length? Just because the old two line function did that, it shouldn't influence how we approach security going forward, in my opinion ...

Googling this also lead me to find this:
https://learn.microsoft.com/en-us/sql/t-sql/statements/create-login-transact-sql?view=sql-server-ver16#password-password
>Passwords can include a-z, A-Z, 0-9, and most non-alphanumeric characters. **_Passwords cannot contain single quotes_**

"_most non-alphanumeric characters_" is so ambigious... @ErikEJ perhaps you can enlighten us? I can't find any comprehensive list of allowed characters.

But what it clearly says is that SQL Server will not allow single quotes in passwords, which is currently a possibility in this password generator PR. 
Should we completely strip single quotes from the generator or should we add a another flag like the `-NoSpecialCharacters`, and call it `-ForSqlServer` and validate it against a different rule set?
It's tempting to do this. But then where goes the line? Should we add flags for all possible systems that have their own quirks and allowed special characters?

If we just remove the single quote, and then we also find out that for an example Azure Cosmos DB doesn't allow `@` characters, then should we also remove that special character ?
We end up with weaker and weaker passwords for every exception we have to make.

Let me know what you think

Reply:
I prefer to use Mansged Identity wherever possible and perfer AAD only with Azure SQL. If I need a password I use  guid string, no risk of invalid characters and quite  random.

Reply:
Solved by adding optional `-AvoidCharacters` parameter.
For SQL server we can add `-AvoidCharacters "'"` to ensure that the generated password does not contain single quotes.